predict_piece.py

```diff
@@ -1,66 +1,79 @@
+from data.quantizer import MidiQuantizer
 from modules.label_smoothing import LabelSmoothing
-from utils import avg_distance, piece_av_files, decode_and_output
+from data.tokenizer import VelocityEncoder, QuantizedMidiEncoder
+from data.dataset import MyTokenizedMidiDataset, quantized_piece_to_records
+from utils import vocab_sizes, piece_av_files, decode_and_output, calculate_average_distance
 
 
 @torch.no_grad()
-def predict_piece_dashboard(cfg: DictConfig):
+def predict_piece_dashboard():
+    dev = torch.device("cuda")
+
     with st.sidebar:
         model_path = st.selectbox(label="model", options=glob.glob("models/*.pt"))
 
-    checkpoint = torch.load(model_path, map_location=cfg.device)
+    checkpoint = torch.load(model_path, map_location=dev)
     params = pd.DataFrame(checkpoint["cfg"]["model"], index=[0])
     train_cfg = OmegaConf.create(checkpoint["cfg"])
 
+    # Prepare everythin required to make inference
+    quantizer = MidiQuantizer(
+        n_dstart_bins=train_cfg.dataset.quantization.dstart,
+        n_duration_bins=train_cfg.dataset.quantization.duration,
+        n_velocity_bins=train_cfg.dataset.quantization.velocity,
+    )
+    src_encoder = QuantizedMidiEncoder(train_cfg.dataset.quantization)
+    tgt_encoder = VelocityEncoder()
+
     st.markdown("Model parameters:")
     st.table(params)
 
-    dev = torch.device(cfg.device)
-    n_dstart_bins, n_duration_bins, n_velocity_bins = train_cfg.dataset.bins.split(" ")
-    hf_dataset = load_dataset(cfg.dataset.dataset_name, split=cfg.dataset_split)
-
-    if cfg.dataset.dataset_name == "roszcz/maestro-v1":
-        pieces_names = zip(hf_dataset["composer"], hf_dataset["title"])
-        with st.sidebar:
-            composer, title = st.selectbox(
-                label="piece",
-                options=[composer + "    " + title for composer, title in pieces_names],
-            ).split("    ")
-
-        one_record_dataset = hf_dataset.filter(lambda x: x["composer"] == composer and x["title"] == title)
-        midi_filename = composer + " " + title + ".mid"
-    else:
-        with st.sidebar:
-            midi_filename = st.selectbox(label="piece", options=[filename for filename in hf_dataset["midi_filename"]])
-        one_record_dataset = hf_dataset.filter(lambda x: x["midi_filename"] == midi_filename)
+    dataset_name = st.text_input(label="dataset", value=train_cfg.dataset_name)
+    split = st.text_input(label="split", value="test")
+    record_id = st.number_input(label="record id", value=0)
+    hf_dataset = load_dataset(dataset_name, split=split)
+
+    # Select one full piece
+    record = hf_dataset[record_id]
+    piece = MidiPiece.from_huggingface(record)
+
+    # And run full pre-processing ...
+    qpiece = quantizer.inject_quantization_features(piece)
+    sequences = quantized_piece_to_records(
+        piece=qpiece,
+        sequence_len=train_cfg.dataset.sequence_len,
+        sequence_step=train_cfg.dataset.sequence_len,
```

Comment:
possibly a bug?
```suggestion
        sequence_step=train_cfg.dataset.sequence_step,
```

Reply:
I changed that name in the config as well :)

Reply:
I meant smth else, but now I realize what was the idea, nevermind :))